pnet_packet/src/dns.rs

```diff
@@ -467,6 +468,133 @@ pub struct DnsResponse {
     pub payload: Vec<u8>,
 }
 
+fn rname_length(packet: &DnsResponsePacket) -> usize {
+    let mut offset = 0;
+    let mut size = 0;
+    loop {
+        let label_len = packet.packet()[offset] as usize;
+        if label_len == 0 {
+            size += 1;
+            break;
+        }
+        if label_len & 0xC0 == 0xC0 {
+            size += 2;
+            break;
+        }
+        size += label_len + 1;
+        offset += label_len + 1;
+    }
+    size
+}
+
+// parse any string in the DNS message, using its unique encoding
+// Call this on DnsPacket, pass the string([u8]) to be decoded
+pub fn dns_parse_name(packet: &DnsPacket, coded_name: &Vec<u8>) -> String {
```

Comment:
Probably just `parse_name()`? No need for the prefix.

Reply:
Accepted.

Reply:
done